src/video/kmsdrm/SDL_kmsdrmvideo.c

```diff
@@ -55,129 +55,106 @@
+get_driindex(void)
 {
-    SDL_bool available = SDL_FALSE;
-    char device[512];
+    int available = -ENOENT;
+    char device[512] = KMSDRM_DRI_PATH;
     int drm_fd;
     int i;
-
-    SDL_snprintf(device, sizeof (device), KMSDRM_DRI_DEVFMT, KMSDRM_DRI_PATH, devindex);
-
-    drm_fd = open(device, O_RDWR | O_CLOEXEC);
-    if (drm_fd >= 0) {
-        if (SDL_KMSDRM_LoadSymbols()) {
-            drmModeRes *resources = KMSDRM_drmModeGetResources(drm_fd);
-            if (resources) {
-                SDL_LogDebug(SDL_LOG_CATEGORY_VIDEO,
-                  KMSDRM_DRI_DEVFMT
-                  " connector, encoder and CRTC counts are: %d %d %d",
-                  KMSDRM_DRI_PATH, devindex,
-                  resources->count_connectors, resources->count_encoders,
-                  resources->count_crtcs);
-
-                if (resources->count_connectors > 0
-                 && resources->count_encoders > 0
-                 && resources->count_crtcs > 0)
-                {
-                    available = SDL_FALSE;
-                    for (i = 0; i < resources->count_connectors; i++) {
-                        drmModeConnector *conn = KMSDRM_drmModeGetConnector(drm_fd,
-                            resources->connectors[i]);
-
-                        if (!conn) {
-                            continue;
-                        }
-
-                        if (conn->connection == DRM_MODE_CONNECTED && conn->count_modes) {
-                            if (SDL_GetHintBoolean(SDL_HINT_KMSDRM_REQUIRE_DRM_MASTER, SDL_TRUE)) {
-                                /* Skip this device if we can't obtain DRM master */
-                                KMSDRM_drmSetMaster(drm_fd);
-                                if (KMSDRM_drmAuthMagic(drm_fd, 0) == -EACCES) {
+    int devindex = -1;
+    DIR *folder = opendir(device);
+    if (!folder) {
+        SDL_SetError("Failed to open directory '%s'.\n", device);
+        return -ENOENT;
+    }
+
+    memcpy(device + KMSDRM_DRI_PATHSIZE, KMSDRM_DRI_DEVNAME,
+           KMSDRM_DRI_DEVNAMESIZE);
+    for (struct dirent *res; (res = readdir(folder));) {
+        if (!memcmp(res->d_name, KMSDRM_DRI_DEVNAME, KMSDRM_DRI_DEVNAMESIZE)) {
+            memcpy(device + KMSDRM_DRI_PATHSIZE + KMSDRM_DRI_DEVNAMESIZE,
+                   res->d_name + KMSDRM_DRI_DEVNAMESIZE, 5);
+
+            drm_fd = open(device, O_RDWR | O_CLOEXEC);
+            if (drm_fd >= 0) {
+                devindex = SDL_atoi(device + KMSDRM_DRI_PATHSIZE +
+                                    KMSDRM_DRI_DEVNAMESIZE);
+                if (SDL_KMSDRM_LoadSymbols()) {
+                    drmModeRes *resources = KMSDRM_drmModeGetResources(drm_fd);
+                    if (resources) {
+                        SDL_LogDebug(SDL_LOG_CATEGORY_VIDEO,
+                                     KMSDRM_DRI_CARDPATHFMT
+                                     " connector, encoder and CRTC counts "
+                                     "are: %d %d %d",
```

Comment:
You don't need to aggressively line wrap, FYI.

Reply:
Addressed in just pushed commit